pkgs/development/python-modules/py-datastruct/default.nix

```
@@ -0,0 +1,30 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  poetry-core,
+}:
+
+buildPythonPackage rec {
+  pname = "py-datastruct";
+  version = "1.1.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "kuba2k2";
+    repo = "datastruct";
+    tag = "v${version}";
+    hash = "sha256-KEIvibGnQnIDMpmodWN2Az/ypc37ZyGvgVPC7voFmlA=";
+  };
+
+  build-system = [ poetry-core ];
+
+  pythonImportsCheck = [ "datastruct" ];
+
```

```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```
To run the unit tests

---

^ @mevatron